pkg/provider/passworddepot/passworddepot_api.go

```diff
@@ -0,0 +1,330 @@
+	Paramstr    string    `json:"paramstr"`
+	Loginid     string    `json:"loginid"`
+	Passid      string    `json:"passid"`
+	Donotaddon  string    `json:"donotaddon"`
+	Markassafe  string    `json:"markassafe"`
+	Safemode    string    `json:"safemode"`
+}
+
+var errDBNotFound = errors.New("Database not found")
+var errSecretNotFound = errors.New("Secret not found")
+
+func NewPasswortDepotApi(baseUrl, username, password, hostPort string, skipVerify bool) (*PasswortDepotApi, error) {
+	api := &PasswortDepotApi{
+		baseUrl:  baseUrl,
+		hostPort: hostPort,
+		username: username,
+		password: password,
+	}
+
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: skipVerify},
+	}
+
+	api.client = &http.Client{Transport: tr}
+	err := api.login()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to login")
+	}
+	return api, nil
+}
+
+func (api *PasswortDepotApi) doAuthenticatedRequest(r *http.Request) (*http.Response, error) {
+	r.Header.Add("access_token", api.secret.AccessToken)
+	r.Header.Add("client_id", api.secret.ClientID)
+
+	return api.client.Do(r)
+}
+func (api *PasswortDepotApi) getDatabaseFingerprint(database string) (string, error) {
+	databases, err := api.ListDatabases()
+	if err != nil {
+		return "", errors.Wrap(err, "error: getting database list")
+	}
+
+	for _, db := range databases.Databases {
+		if strings.Contains(db.Name, database) {
+			return db.Fingerprint, nil
+		}
+	}
+
+	return "", errDBNotFound
+}
+
+func (api *PasswortDepotApi) getSecretFingerprint(databaseFingerprint, secretName string, folder string) (string, error) {
+	secrets, err := api.ListSecrets(databaseFingerprint, folder)
+	if err != nil {
+		return "", errors.Wrap(err, "error: getting secrets list")
+	}
+
+	parts := strings.Split(secretName, ".")
+	searchName := parts[0]
+	var fingerprint string
+	for _, entry := range secrets.Entries {
+		if strings.Contains(entry.Name, searchName) {
+			fingerprint = entry.Fingerprint
+			if len(parts) > 1 {
+				return api.getSecretFingerprint(databaseFingerprint, strings.Join(parts[1:], "."), fingerprint)
+			} else {
+				return fingerprint, nil
+			}
+		}
+	}
+
+	return "", errSecretNotFound
+}
+
+func (api *PasswortDepotApi) getEndpointUrl(endpoint string) string {
+	return fmt.Sprintf("https://%s:%s/v1.0/%s", api.baseUrl, api.hostPort, endpoint)
+}
+
+func (api *PasswortDepotApi) login() error {
+	loginRequest, err := http.NewRequest("GET", api.getEndpointUrl("login"), nil)
+	if err != nil {
+		return errors.Wrap(err, "error creating request")
+	}
+	loginRequest.Header.Add("user", api.username)
+	loginRequest.Header.Add("pass", api.password)
+
+	resp, err := api.client.Do(loginRequest)
+	if err != nil {
+		return errors.Wrap(err, "error: do request")
+	}
+	defer func() {
+		if resp.Body != nil {
+			resp.Body.Close()
+		}
+	}()
+
+	var buf bytes.Buffer
+	_, err = buf.ReadFrom(resp.Body)
+	if err != nil {
+		return errors.Wrap(err, "error: read body")
+	}
+
+	if resp.StatusCode > 399 {
```

Comment:
I think this should be `if resp.StatusCode < 200 || resp.StatusCode > 299` right?

Reply:
You are right!